tools/ogcProcess_otb_bandmath/OTB_BandMath.xml

```diff
@@ -0,0 +1,91 @@
+<tool id="otb_band_math" name="OTB.BandMath" version="1.0" profile="22.05">
+    <description> outputs a monoband image as a result of a mathematical operation on several multi-band images</description>
+    <requirements>
+        <requirement type="package" version="4.3.1">r-base</requirement>
+        <requirement type="package" version="1.20.4">r-getopt</requirement>
+        <requirement type="package" version="0.2.3">r-httr2</requirement>
+        <requirement type="package" version="1.8.7">r-jsonlite</requirement>
+    </requirements>
+    <command detect_errors='exit_code'>
+        <![CDATA[
+            Rscript '$__tool_directory__/OTB_BandMath.R' 
+                --file '$file' 
+                --processingMemory '$processing_memory' 
+                --mathExpression '$math_expression'
+                --outputImage '$output_image' 
+                --outputType '$output_type' 
+                --outputFormat '$output_format' 
+                --outputData '$output_data'    
+        ]]>
+    </command>
+    <inputs>
+        <param type="data" format="txt" name="file" label="List of images" help="Image-list of operands to the mathematical expression (.txt file, one per line)" />
+        <param type="integer" name="processing_memory" label="Random-Access Memory (RAM)" value="256" optional="true" help="Available memory for processing (in MB)" />
+        <param type="text" name="math_expression" label="muParser" optional="false" help="The muParser mathematical expression to apply on input images (e.g., 'im1b3,im1b2,im1b1'). See 'Help' for documentation."/>
+        <param type="select" name="output_image" label="Output image" help="Output image which is the result of the mathematical expressions on input image-list operands">
+            <option value="float">float</option>
+            <option value="uint8">uint8</option>
+            <option value="uint16">uint16</option>
+            <option value="int16">int16</option>
+            <option value="int32">int32</option>
+            <option value="double">double</option>
+        </param>
+        <param type="select" name="output_type" label="Output image format" help="Select .png for obtaining the response as an OGC web service. Select .tiff or .jpeg for using it in a workflow.">
+            <option value="png">.png (for OGC web services)</option>
+            <option value="tiff">.tiff (for workflows)</option>
```

Comment:
I don't understand why tiff is good for workflow but png not?

Reply:
Ah do you mean if you want to pass it along to the next tool it strictly needs to be in tiff and jpeg, so the URL needs to be like that?

If so maybe we should make the output_format a conditional and offer different output_type options?

Reply:
Right, if the tool should be used as input for another process, it has to be tiff or jpeg. Do you mean conditional by asking the user if the tool should be used in a workflow?

Reply:
Something like that:

```
<conditional name="outputs">
    <param type="select" name="output_format" label="Download the result to your Galaxy history or get the URL?" help="Select 'Get URL' to use the result as input for another OGC API Process in a workflow.">
        <option value="download">Download</option>
        <option value="getUrl">Get URL</option>
    </param>
    <when value="download">
        <param type="select" name="output_type" label="Output image format" help="Select .png for obtaining the response as an OGC web service. Select .tiff or .jpeg for using it in a workflow.">
            <option value="png">.png (for OGC web services)</option>
            <option value="tiff">.tiff (for workflows)</option>
            <option value="jpeg">.jpeg (for workflows)</option>
        </param>    </when>
    <when value="getURL">
        <param type="select" name="output_type" label="Output image format" help="Select .png for obtaining the response as an OGC web service. Select .tiff or .jpeg for using it in a workflow.">
            <option value="tiff">.tiff (for workflows)</option>
            <option value="jpeg">.jpeg (for workflows)</option>
        </param>
    </when>
</conditional>        
```

Reply:
If we do tools manually we can do those things and prevent user errors, I guess we can not do that when we convert automatically.

Reply:
Mh, but what if a user wants to get the URL to the png file? With this conditional, this would not be possible. GetUrl is needed if the user wants to create a workflow but there might be cases other than creating workflows where the user wants the URL.